script/demoScripts/demoGarden.ts

```diff
@@ -0,0 +1,341 @@
+        const swapDataItem = await getUniswapDataERC20toExactERC20(
+          UNISWAP_ADDRESS,
+          1, // Mainnet chain ID
+          WETH_ADDRESS,
+          USDC_ADDRESS,
+          amountInWei, // Exact USDC output we want
+          LIFI_ADDRESS,
+          true, // requiresDeposit
+          Math.floor(Date.now() / 1000) + 60 * 60 // 1 hour deadline
+        )
+
+        swapData = [swapDataItem]
+        inputAmount = BigNumber.from(swapDataItem.fromAmount) // The amount of WETH needed (with slippage)
+
+        consola.info(
+          `Swap prepared: ${utils.formatUnits(
+            inputAmount,
+            18
+          )} WETH (max with slippage) -> ${amountInUsdc} USDC (exact)`
+        )
+
+        // Check WETH balance and approval
+        consola.info('\n💰 Checking WETH balance and approval...')
+        await ensureBalanceAndAllowanceToDiamond(
+          WETH_ADDRESS,
+          wallet,
+          LIFI_ADDRESS,
+          inputAmount,
+          false
+        )
+      }
+
+      // Step 1: Get quote from Garden API
+      consola.info('\n📡 Fetching quote from Garden API...')
+
+      const quoteUrl = `https://api.garden.finance/v2/quote?from=ethereum:usdc&to=base:usdc&from_amount=${amountInWei.toString()}`
+      const response = await fetch(quoteUrl)
+
+      if (!response.ok)
+        throw new Error(`Failed to fetch quote: ${response.statusText}`)
+
+      const quote: IGardenQuote = await response.json()
+
+      if (quote.status !== 'Ok' || !quote.result || quote.result.length === 0)
+        throw new Error('No valid quote received from Garden API')
+
+      const selectedQuote = quote.result[0]
+      if (!selectedQuote) throw new Error('No quote selected')
+
+      consola.success(`Quote received:`)
+      consola.info(
+        `  Input: ${selectedQuote.source.display} ${selectedQuote.source.asset
+          .split(':')[1]
+          ?.toUpperCase()}`
+      )
+      consola.info(
+        `  Output: ${
+          selectedQuote.destination.display
+        } ${selectedQuote.destination.asset.split(':')[1]?.toUpperCase()}`
+      )
+      consola.info(`  Solver: ${selectedQuote.solver_id}`)
+
+      // Step 2: Submit order to Garden API
+      consola.info('\n📝 Submitting order to Garden...')
+
+      const orderPayload = {
+        source: {
+          asset: 'ethereum:usdc',
+          owner: address,
+          amount: amountInWei.toString(),
+        },
+        destination: {
+          asset: 'base:usdc',
+          owner: address, // Same address on destination chain
+          amount: selectedQuote.destination.amount,
+        },
+      }
+
+      const orderResponse = await fetch(
+        'https://api.garden.finance/v2/orders',
+        {
+          method: 'POST',
+          headers: {
+            'garden-app-id':
+              '7648702b1997e55a3763afa5dd7ace3d4bd23348ee0423cc27a18ef3e28cb2b7', // [pre-commit-checker: not a secret]
```

Comment:
what is this ID?
Could be a variable and also a comment could help to understand what this is and how it's supposed to be used (always the same? or will the BE need their own? etc.)

Reply:
fixed: cdb014dbb4c5855724f90ceb291087487ea083d4